### PR TITLE
Use NonZeroUsize everywhere 

### PR DESCRIPTION
**Pull request by yshui**
This can save 16 bytes per Entry<T>, which can be significant if T is
small.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -20,6 +20,77 @@ use std::{
...
+  }
+
+  /// Create a new index from a usize, without checking if `index` is `usize::MAX`.
+  ///
+  /// # Safety
+  ///
+  /// `index` must not be `usize::MAX`
+  #[inline]
+  const unsafe fn new_unchecked(index: usize) -> Self {
+    Self(unsafe { NonZeroUsize::new_unchecked(index.wrapping_add(1)) })
```
Comment (sgodwincs) [inline src/lib.rs:57]: Is the wrapping_add needed here? It seems we can just do `i + 1` since it will be undefined behavior either way?

**Comment by sgodwincs** [comment]
Once you submit the previous PR, I'll take another look at this so it's not so noisy

File: src/lib.rs (hunk 1 above)
Comment (yshui) [reply src/lib.rs:57]: Good point! I believe you are right.

**Comment by yshui** [comment]
btw, can we plan a new release?

File: src/lib.rs [hunk 2]
```diff
@@ -1226,31 +1310,24 @@ impl<T> VecList<T> {
...
-  fn set_vacant_head(&mut self, index: Option<usize>) {
-    self.vacant_head =
-      index.map(|index| NonZeroUsize::new(index + 1).expect("vacant head should not be 0"));
+  fn set_vacant_head(&mut self, index: Option<NonMaxUsize>) {
+    self.vacant_head = index;
   }
 
   /// Convenience function for returning the actual tail index.
   #[must_use]
   fn tail(&self) -> Option<usize> {
```
Comment (sgodwincs) [inline src/lib.rs:1329]: Same with this one

File: src/lib.rs [hunk 3]
```diff
@@ -1226,31 +1310,24 @@ impl<T> VecList<T> {
   }
 
   /// Convenience function for returning setting the offset head index.
-  fn set_head(&mut self, index: usize) {
-    self.head = Some(NonZeroUsize::new(index + 1).expect("head should not be 0"));
+  fn set_head(&mut self, index: NonMaxUsize) {
```
Comment (sgodwincs) [inline src/lib.rs:1313]: set_head, set_tail, and set_vacant_head should probably just be inlined at this point instead of separate functions.

**Review by sgodwincs — CHANGES_REQUESTED** [review]
Can you add #[must_use] to all new functions that return a value and have no side-effects?

**Comment by sgodwincs** [comment]
I'll release 0.4.1 after this is submitted.

**Review by sgodwincs — APPROVED** [review]

**Pull request merged**